### ISSUE TITLE
TypeScript Strict mode breaks build

### ISSUE THREAD

**Issue by rodzappa** [state-adapt/state-adapt#1]
when strict mode is enable (
```
...
"compilerOptions": {
    ...
    "strict": true,
    ...
}
...
```
in tsconfig.base.json the IDE triggers errors and the build of ng-sa-ngrx-counter app breaks:

First Error:
```
TS2322: Type '(state: AdaptModel | undefined, action: PatchState) => AdaptModel' is not assignable to type 'ActionReducer<AdaptModel, Action>'.   Types of parameters 'action' and 'action' are incompatible.     Type 'Action' is missing the following properties from type 'PatchState': source, payload
```

Second Error:
```
TS2322: Type '(action: CommonAction & PatchState) => (CommonAction & PatchState) | { action: undefined; actionType: undefined; source: undefined; payload: any; type: string; }' is not assignable to type 'ActionSanitizer'.   Types of parameters 'action' and 'action' are incompatible.     Type 'Action' is not assignable to type 'CommonAction & PatchState'.       Type 'Action' is missing the following properties from type 'PatchState': source, payload
```

The issue occurs in app.module.ts, at @NgModule annotation:

```
@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    CounterDemoModule,
    StoreModule.forRoot({ adapt: adaptReducer }),  //  <-- first error here!
    StoreDevtoolsModule.instrument({
      maxAge: 25,
      logOnly: environment.production,
      actionSanitizer, //  <-- second error here!
      stateSanitizer,
    }),
  ],
  bootstrap: [AppComponent],
})
```

Steps to reproduce:

1. Clone repo (main branch)
2. Add  "strict": true, into CompilerOptions inside tsconfig.base.json
3. Observe the IDE starts alarming.

Workaround: Not known.

**Comment by mfp22** [comment]
@rodzappa Thanks for pointing this out! I originally developed this as part of a non-strict TS codebase, so I need to turn it on here and go through all the apps and make some fixes.

**Comment by mfp22** [comment]
Fixed with 2bd5a8eef27fb1de2cd62dbb665d3eb84befb980
